src/main/java/io/nats/client/impl/NatsMessageConsumerBase.java

```diff
@@ -101,21 +101,19 @@ public void stop() {
 
     @Override
     public void close() throws Exception {
-        lenientClose();
+        stopped.set(true);
+        shutdownSub();
     }
 
-    protected void finishAndClose() {
-        if (pmm != null) {
-            pmm.shutdownHeartbeatTimer();
-        }
+    protected void fullClose() {
+        stopped.set(true);
         finished.set(true);
-        lenientClose();
+        shutdownSub();
     }
 
-    protected void lenientClose() {
+    protected void shutdownSub() {
```

Comment:
Being a good citizen and shutting down the sub and the heartbeat timer.
1. Called when completely shutting down close()
2. Called during endless consumes to stop the current sub since a new one is being made.